lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -401,11 +401,12 @@ def bounce_service host, service, curl_retries = nil, port = nil
         # @param [Block] block This method will yield to a block of code passed
         #                      by the caller; this can be used for additional
         #                      validation, etc.
+        # @param [Boolean] run_in_parallel Whether to run on each host in parallel.
         #
         # @return [Array<Result>, Result, nil] An array of results, a result object,
         #   or nil. Check {#run_block_on} for more details on this.
-        def apply_manifest_on(host, manifest, opts = {}, &block)
-          block_on host do | host |
+        def apply_manifest_on(host, manifest, opts = {}, run_in_parallel=false, &block)
```

Comment:
Yeah, I think it has to be.  The main opts hash is typically passed around Beaker from method to method.  If you want to be able to execute the same method in parallel or not in parallel depending on the context, it pretty much needs to be outside of the options hash, unless I'm misunderstanding something.